### DISCUSSION TITLE
SysML v2 textual format out of SysON

### DISCUSSION THREAD

**Discussion by MFK-18** [eclipse-syson/syson#683] [Q&A] [answered]
Hello everyone, 

the question is as stated: how can i convert my model from graphical to textual format or am I missing something obvious. I know I can export the model but then I have only the JSON files. 

I could find all for Import via SysIDE but I couldnt find the info for exporting a model creating in SysOn to textual.

Thanks for any help.

**Comment by stephanelacrampe** [comment]
I think this is what you're looking for:
![textual export](https://github.com/user-attachments/assets/a9e176f5-c4a5-4675-8cd6-1959ca86d530)

@AxelRICHARD  looks like the documentation is wrong there as it states it gives you a Json file  https://doc.mbse-syson.org/syson/v2024.7.0/user-manual/hands-on/how-tos/model-management.html#download-model

**Reply by MFK-18** [reply]
Thank you very much! Now there is a problem still with it for my project. It always gives me an empty output. 
The following model:
![image](https://github.com/user-attachments/assets/5b2f3b07-6f2a-4e53-a6a4-25db52e84e36)
gives: 
![image](https://github.com/user-attachments/assets/7d4bef5e-636b-43f5-8c7e-0c983ea54299)


could this because of creating the model some time ago and since then some updates made some connections disappear? 

Thanks for any help in advance again!

**Reply by AxelRICHARD** [reply]
Hello,
@MFK-18 It's quite simple, If you want your model to be exported as a SysMLv2 textual file you have to suffix your document with ".sysml".
In your screenshot, it should be `ARCH4SysMLv2.sysml` instead of `ARCH4SysMLv2`.
@stephanelacrampe I will update this section of the documentation, thank you for pointing this.
Regards,

**Reply by MFK-18** [reply]
Sadly the problem is the exported data is always empty, if I choose the standard batman project or create a new one the export works fine, but for the existing project ARCH4SysMLv2 the export is always just empty. It shouldnt because there are many objects already created and connected.

**Reply by AxelRICHARD** [reply, marked as answer]
From your screenshot, It seems you have created objects outside the implicit root Namespace. All these elements created outside the root Namespace won't be exported.
If you create some objects under the `Namespace` element, you should see them exported.
That's a problem we have fixed very recently (https://github.com/eclipse-syson/syson/issues/671) and will be available in the next release.
Sorry for this problem, but the only way to fix that is to currently create again all your elements under the `Namespace` element.
Regards,
